portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -16,61 +16,85 @@
 
 import java.io.IOException;
 import java.util.*;
-import com.google.appengine.api.datastore.DatastoreService;
-import com.google.appengine.api.datastore.DatastoreServiceFactory;
-import com.google.appengine.api.datastore.Entity;
-import com.google.appengine.api.datastore.PreparedQuery;
-import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.*;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 import javax.servlet.RequestDispatcher;
 import com.google.gson.*;
 
-/** Servlet that returns some example content. */
+/** Servlet that returns comment data created by the user. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int numComments = 0;
+
+    try {
+        numComments = Integer.parseInt(request.getParameter("number"));
+    } catch (NumberFormatException e) {
+        // Leave the default value alone.
+    }
+    
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
-    int numComments = Integer.parseInt(request.getParameter("number"));
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    ArrayList<String> comments = new ArrayList<String>();
+    /* If no comments are returned by the query, set a descriptive HTTP status code to signify an error. */
+    int size = results.countEntities(FetchOptions.Builder.withLimit(1000));
+    if (size == 0) {
+        response.sendError(404);
+        return;
+    }
+
+    HashMap<String, String> commentData = new HashMap<String, String>();
     Iterator<Entity> iter = results.asIterable().iterator();
```

Comment:
This line ought to be going away given the comments above, but I do want to mention that [`PreparedQuery#asIterator`](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery.html#asIterator--) is a more concise way of writing this.